build: configure doesn't pickup Qt if depends is built against libc++

On Linux, using the master branch (3e306ee1d5c79eda64eb991fb2696bf530f0b30d) or the 0.21 one (926f76cb205ce9ef085f4945cdb439746b140414), after building depends against `libc++`:
```
$ make -j 9 -C depends CC=clang CXX='clang++ -stdlib=libc++'
```
configuring with
```
$ ./autogen.sh && CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site ./configure -q CC=clang CXX='clang++ -stdlib=libc++'
```
fails to pick up the Qt package.

This issue prevents TSAN from being enabled for GUI builds (see #19162).

From `config.log`:
```
configure:25012: checking for QMinimalIntegrationPlugin (-lqminimal)
configure:25030: clang++ -stdlib=libc++ -std=c++17 -o conftest -fPIC -pipe -O2  -DQT_NETWORK_LIB -DQT_CORE_LIB -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtNetwork -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtCore -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtWidgets -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtGui -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtCore -DQT_GUI_LIB -DQT_CORE_LIB -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtGui -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtCore -DQT_CORE_LIB -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/QtCore  -I/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib  conftest.cpp -lqminimal -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L//home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L//home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5XcbQpa /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5ServiceSupport.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5ThemeSupport.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5FontDatabaseSupport.a -lfontconfig -lfreetype /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5EdidSupport.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libxcb-static.a -lxcb -lxkbcommon-x11 -lxkbcommon -ldl -lQt5ServiceSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5DBus /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5ThemeSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lQt5FontDatabaseSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lfontconfig -lfreetype -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5EdidSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5ServiceSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5DBus /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L//home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L//home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5XcbQpa /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5ServiceSupport.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5ThemeSupport.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5FontDatabaseSupport.a -lfontconfig -lfreetype /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5EdidSupport.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libxcb-static.a -lxcb -lxkbcommon-x11 -lxkbcommon -ldl -lQt5ServiceSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5DBus /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5ThemeSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lQt5FontDatabaseSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lfontconfig -lfreetype -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5EdidSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5ThemeSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5DBus.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L//home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5FontDatabaseSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lfontconfig -lfreetype -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5FbSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5EventDispatcherSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5EdidSupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5DeviceDiscoverySupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5AccessibilitySupport /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5Network /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5Widgets /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5Gui /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -lpthread /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtlibpng.a /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtharfbuzz.a -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib -lQt5Core -lpthread -lm /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libqtpcre2.a -ldl  -L/home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/plugins/platforms  >&5
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a(qtextengine.o): in function `void std::vector<int, std::allocator<int> >::_M_realloc_insert<int const&>(__gnu_cxx::__normal_iterator<int*, std::vector<int, std::allocator<int> > >, int const&)':
qtextengine.cpp:(.text._ZNSt6vectorIiSaIiEE17_M_realloc_insertIJRKiEEEvN9__gnu_cxx17__normal_iteratorIPiS1_EEDpOT_[_ZNSt6vectorIiSaIiEE17_M_realloc_insertIJRKiEEEvN9__gnu_cxx17__normal_iteratorIPiS1_EEDpOT_]+0x143): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Gui.a(qtextengine.o): in function `QTextEngine::insertionPointsForLine(int)':
qtextengine.cpp:(.text._ZN11QTextEngine22insertionPointsForLineEi+0x239): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(qrandom.o): in function `void std::vector<unsigned int, std::allocator<unsigned int> >::_M_realloc_insert<unsigned int>(__gnu_cxx::__normal_iterator<unsigned int*, std::vector<unsigned int, std::allocator<unsigned int> > >, unsigned int&&)':
qrandom.cpp:(.text._ZNSt6vectorIjSaIjEE17_M_realloc_insertIJjEEEvN9__gnu_cxx17__normal_iteratorIPjS1_EEDpOT_[_ZNSt6vectorIjSaIjEE17_M_realloc_insertIJjEEEvN9__gnu_cxx17__normal_iteratorIPjS1_EEDpOT_]+0x143): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(qthread.o): in function `QThreadCreateThread::run()':
qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0x72): undefined reference to `std::__atomic_futex_unsigned_base::_M_futex_wait_until(unsigned int*, unsigned int, bool, std::chrono::duration<long, std::ratio<1l, 1l> >, std::chrono::duration<long, std::ratio<1l, 1000000000l> >)'
/usr/bin/ld: qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0x9f): undefined reference to `std::__exception_ptr::operator==(std::__exception_ptr::exception_ptr const&, std::__exception_ptr::exception_ptr const&)'
/usr/bin/ld: qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0xa9): undefined reference to `std::__exception_ptr::exception_ptr::~exception_ptr()'
/usr/bin/ld: qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0x165): undefined reference to `std::__throw_future_error(int)'
/usr/bin/ld: qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0x170): undefined reference to `std::__exception_ptr::exception_ptr::exception_ptr(std::__exception_ptr::exception_ptr const&)'
/usr/bin/ld: qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0x178): undefined reference to `std::rethrow_exception(std::__exception_ptr::exception_ptr)'
/usr/bin/ld: qthread.cpp:(.text._ZN19QThreadCreateThread3runEv[_ZN19QThreadCreateThread3runEv]+0x195): undefined reference to `std::__exception_ptr::exception_ptr::~exception_ptr()'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(qthread_unix.o):(.data.rel.local.DW.ref._ZTIN10__cxxabiv115__forced_unwindE[DW.ref._ZTIN10__cxxabiv115__forced_unwindE]+0x0): undefined reference to `typeinfo for __cxxabiv1::__forced_unwind'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(double-conversion.o): in function `bool double_conversion::(anonymous namespace)::ConsumeSubString<unsigned short const*>(unsigned short const**, unsigned short const*, char const*, bool)':
double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKtEEbPT_S4_PKcb[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0xc9): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKtEEbPT_S4_PKcb[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0xd1): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKtEEbPT_S4_PKcb[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0x15d): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKtEEbPT_S4_PKcb[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0x165): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(double-conversion.o): in function `bool double_conversion::(anonymous namespace)::ConsumeSubString<char const*>(char const**, char const*, char const*, bool)':
double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKcEEbPT_S4_S3_b[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0xc3): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKcEEbPT_S4_S3_b[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0xcb): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKcEEbPT_S4_S3_b[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0x145): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZN17double_conversion12_GLOBAL__N_1L16ConsumeSubStringIPKcEEbPT_S4_S3_b[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0x14d): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(double-conversion.o): in function `double double_conversion::StringToDoubleConverter::StringToIeee<char const*>(char const*, int, bool, int*) const':
double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0x8da): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0x8e2): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0x92b): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKcEEdT_ibPi]+0x933): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: /home/hebasto/GitHub/bitcoin/depends/x86_64-pc-linux-gnu/lib/libQt5Core.a(double-conversion.o): in function `double double_conversion::StringToDoubleConverter::StringToIeee<unsigned short const*>(unsigned short const*, int, bool, int*) const':
double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0xa43): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0xa4b): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
/usr/bin/ld: double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0xa94): undefined reference to `std::locale::classic()'
/usr/bin/ld: double-conversion.cc:(.text._ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi[_ZNK17double_conversion23StringToDoubleConverter12StringToIeeeIPKtEEdT_ibPi]+0xa9c): undefined reference to `std::ctype<char> const& std::use_facet<std::ctype<char> >(std::locale const&)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure:25030: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "Bitcoin Core"
| #define PACKAGE_TARNAME "bitcoin"
| #define PACKAGE_VERSION "21.99.0"
| #define PACKAGE_STRING "Bitcoin Core 21.99.0"
| #define PACKAGE_BUGREPORT "https://github.com/bitcoin/bitcoin/issues"
| #define PACKAGE_URL "https://bitcoincore.org/"
| #define HAVE_CXX17 1
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| #define USE_ASM 1
| #define ENABLE_SSE41 1
| #define ENABLE_AVX2 1
| #define ENABLE_SHANI 1
| #define HAVE___INT128 1
| #define HAVE_PTHREAD_PRIO_INHERIT 1
| #define HAVE_PTHREAD 1
| #define HAVE_DECL_STRERROR_R 1
| #define HAVE_STRERROR_R 1
| #define STRERROR_R_CHAR_P 1
| #define HAVE_ENDIAN_H 1
| #define HAVE_BYTESWAP_H 1
| #define HAVE_STDIO_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_UNISTD_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_SYS_SELECT_H 1
| #define HAVE_SYS_PRCTL_H 1
| #define HAVE_SYS_SYSCTL_H 1
| #define HAVE_DECL_GETIFADDRS 1
| #define HAVE_DECL_FREEIFADDRS 1
| #define HAVE_DECL_STRNLEN 1
| #define HAVE_DECL_FORK 1
| #define HAVE_DECL_SETSID 1
| #define HAVE_DECL_PIPE2 1
| #define HAVE_DECL_LE16TOH 1
| #define HAVE_DECL_LE32TOH 1
| #define HAVE_DECL_LE64TOH 1
| #define HAVE_DECL_HTOLE16 1
| #define HAVE_DECL_HTOLE32 1
| #define HAVE_DECL_HTOLE64 1
| #define HAVE_DECL_BE16TOH 1
| #define HAVE_DECL_BE32TOH 1
| #define HAVE_DECL_BE64TOH 1
| #define HAVE_DECL_HTOBE16 1
| #define HAVE_DECL_HTOBE32 1
| #define HAVE_DECL_HTOBE64 1
| #define HAVE_DECL_BSWAP_16 1
| #define HAVE_DECL_BSWAP_32 1
| #define HAVE_DECL_BSWAP_64 1
| #define HAVE_BUILTIN_CLZL 1
| #define HAVE_BUILTIN_CLZLL 1
| #define HAVE_MALLOC_INFO 1
| #define HAVE_MALLOPT_ARENA_MAX 1
| #define HAVE_POSIX_FALLOCATE 1
| #define HAVE_DEFAULT_VISIBILITY_ATTRIBUTE 1
| #define HAVE_THREAD_LOCAL 1
| #define HAVE_GMTIME_R 1
| #define HAVE_SYS_GETRANDOM 1
| #define HAVE_GETENTROPY 1
| #define HAVE_GETENTROPY_RAND 1
| #define HAVE_FDATASYNC 1
| #define HAVE_O_CLOEXEC 1
| #define HAVE_WEAK_GETAUXVAL 1
| #define HAVE_STD__SYSTEM 1
| #define HAVE_SYSTEM HAVE_STD__SYSTEM || HAVE_WSYSTEM
| #define QT_STATICPLUGIN 1
| /* end confdefs.h.  */
| 
|       #include <QtPlugin>
|       Q_IMPORT_PLUGIN(QMinimalIntegrationPlugin)
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:25034: result: no
configure:25044: error: QMinimalIntegrationPlugin not found.
```

--- a/depends/packages/qt.mk
+++ b/depends/packages/qt.mk
@@ -140,7 +140,11 @@ $(package)_config_opts_linux += -no-feature-vulkan
 $(package)_config_opts_linux += -dbus-runtime
 $(package)_config_opts_arm_linux += -platform linux-g++ -xplatform bitcoin-linux-g++
 $(package)_config_opts_i686_linux  = -xplatform linux-g++-32
+ifneq (,$(findstring -stdlib=libc++,$($(1)_cxx)))
+$(package)_config_opts_x86_64_linux = -xplatform linux-clang-libc++
+else
 $(package)_config_opts_x86_64_linux = -xplatform linux-g++-64
+endif
 $(package)_config_opts_aarch64_linux = -xplatform linux-aarch64-gnu-g++
 $(package)_config_opts_powerpc64_linux = -platform linux-g++ -xplatform bitcoin-linux-g++
 $(package)_config_opts_powerpc64le_linux = -platform linux-g++ -xplatform bitcoin-linux-g++

